hackforla/peopledepot
# Pre-commit CI

- Fixes #124

### What changes did you make?

- adapted pre-commit config for ci
- updated pre-commit config
- updated and extended documentation for pre-commit
- formatted documentation touched by pre-commit checks
- removed local linters and fomatters

### Why did you make the changes (we will use this info to test)?

- We're using pre-commit for linting and formatting, so there's no need to also install the packages locally. Pre-commit manages its own environment that's separate from the project.
- I had to switch some pre-commit hooks to ones that will also download the tools rather than rely on them being present in the local system, since they won't be available in the ci environment.
- Some hooks are just outdated so I updated them.
- There's a few new hooks that I thought were good to add.
- I updated the documentation both to fix formatting and with content for the current changes.

### Before and After

I created test PRs to show the CI service at work.
[Before](https://github.com/fyliu/peopledepot/pull/15): this branch adds a squashed commit of the changes made in [initial-data-migration-36](https://github.com/hackforla/peopledepot/tree/initial-data-migration-36). It has some linting problems that need fixing.

[After](https://github.com/fyliu/peopledepot/pull/16): this is after fixing the linter issues. It is one more commit on top of the Before branch.

### How to test

You will have to connect to the pre-commit.ci service in your fork in order for this to work in your fork. Just create PRs from the branches in my fork (or anywhere) to see it working.

I haven't connected it to the hackforla repo yet since the current pre-commit config we have doesn't quite work for ci.
